suseconnect/migration.go

```diff
@@ -0,0 +1,653 @@
+package main
+
+// TODO LIST
+// * zypp_backup/zypp_restore functions
+// * --selfupdate option
+// * --query plugin option
+// * --break-my-system option
+// * --product option (offline migration)
+// * obsolete repo disabling (including --disable-repos option)
+// * interactive migration mode
+// * snapshots (snapper wrapper)
+// * Leap -> SLES migration case
+// * system.execute() function with pipe output from executed program to stdout
+
+import (
+	_ "embed"
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/signal"
+	"sort"
+	"strings"
+	"syscall"
+
+	"github.com/SUSE/connect-ng/internal/connect"
+)
+
+var (
+	//go:embed migrationUsage.txt
+	migrationUsageText string
+	// flag indicating interuption by INT/TERM signal
+	interrupted bool
+
+	// ErrInterrupted is returned when execution was interrupted by INT/TERM signal
+	ErrInterrupted = errors.New("Interrupted")
+)
+
+// logger shortcuts
+var (
+	Debug      *log.Logger = connect.Debug
+	QuietOut   *log.Logger = connect.QuietOut
+	VerboseOut             = log.New(io.Discard, "", 0)
+)
+
+// this Value type is used to hold values of args which could be passed multiple times
```

Comment:
implements the flag.Value interface ... used to hold ...

Reply:
done